backend/src/datasource.js

```diff
@@ -0,0 +1,67 @@
+import { DataSource } from "apollo-datasource";
+import crypto from "crypto";
+
+export class Post {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString("hex");
+        this.voters = new Map();
+        Object.assign(this, data);
+    }
+}
+export class User {
+    constructor(name) {
+        this.name = name;
```

Comment:
In comparsion to [here](https://github.com/Systems-Development-and-Frameworks/lichtow/pull/8/files#diff-10d67b04ad805bf8ca07ecb8e8488f88d1470956bcb26e612b6b7a940a3fd553R8), you can actually tell that `User` objects ought to have a `name`!